PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['pos.config']
-    _inherit = ['pos.printer']
-    _inherit = ['res.config.settings']

Total Changes: 7

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_epson_printer/models/pos_config.py
+++ b/addons/pos_epson_printer/models/pos_config.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     epson_printer_ip = fields.Char(string='Epson Printer IP', help="Local IP address of an Epson receipt printer.")

--- a/addons/pos_epson_printer/models/pos_printer.py
+++ b/addons/pos_epson_printer/models/pos_printer.py
@@ -6,8 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class PosPrinter(models.Model):
-
-    _inherit = ['pos.printer']
+    _inherit = 'pos.printer'
 
     printer_type = fields.Selection(selection_add=[('epson_epos', 'Use an Epson printer')])
     epson_printer_ip = fields.Char(string='Epson Printer IP Address', help="Local IP address of an Epson receipt printer.", default="0.0.0.0")

--- a/addons/pos_epson_printer/models/res_config_settings.py
+++ b/addons/pos_epson_printer/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     pos_epson_printer_ip = fields.Char(compute='_compute_pos_epson_printer_ip', store=True, readonly=False)
 
